pulsar-broker/src/main/java/com/yahoo/pulsar/broker/PulsarService.java

```diff
@@ -133,11 +139,49 @@ public PulsarService(ServiceConfiguration config) {
         this.webServiceAddressTls = webAddressTls(config);
         this.brokerServiceUrl = brokerUrl(config);
         this.brokerServiceUrlTls = brokerUrlTls(config);
+        this.brokerVersion = fixVersionString(getPropertyFromResource(RESOURCE_NAME, "version"));
         this.config = config;
         this.shutdownService = new MessagingServiceShutdownHook(this);
         loadManagerExecutor = Executors.newSingleThreadScheduledExecutor();
     }
 
+    // If the version string does not contain a patch version, add one so the
+    // version becomes valid according to the SemVer library (see https://github.com/zafarkhaja/jsemver).
+    // This method (and it's only call above in the ctor) may be removed when SemVer accepts null patch versions
+    public static String fixVersionString(String version) {
+        if ( null == version ) {
+            return null;
+        }
+
+        Pattern majorMinorPatchPattern = Pattern.compile("([1-9]+[0-9]*)\\.([1-9]+[0-9]*)\\.([1-9]+[0-9]*)(.*)");
```

Comment:
This could be saved as `private static final ..` (not that it makes a big difference though since it should be only called once.